### PR TITLE
Update to geoserver 2.25.1

### PR DESCRIPTION
**Pull request by landryb**
once all extensions are locally built, this builds, packages & starts fine.

at runtime trying to go to the openlayers preview of a layer barfs on the SLD parsing:
```
Caused by: org.xml.sax.SAXException: Entity resolution disallowed for null
        at org.geoserver.util.AllowListEntityResolver.resolveEntity(AllowListEntityResolver.java:176)
        at org.geoserver.util.AllowListEntityResolver.getExternalSubset(AllowListEntityResolver.java:119)
        at org.apache.xerces.util.EntityResolver2Wrapper.getExternalSubset(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentScannerImpl$ContentDispatcher.resolveExternalSubsetAndRead(Unknown Source)
        at org.apache.xerces.impl.XMLNSDocumentScannerImpl$NSContentDispatcher.scanRootElementHook(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
        at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
        at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
        at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
        at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
        at org.geotools.xml.styling.SLDParser.parseSLD(SLDParser.java:469)
```
i suppose that's related to the new entity resolving checks, but even if globally disabling the url checker it fails. Will have to check if existing SLDs have to be tweaked/modified..

discussed with @f-necas and we think once #4205 is merged to `master`, and georchestra 24.0 branch is created we should aim to update to gs 2.25 in `master`

### CODE REVIEW

**Comment by landryb** [comment]
PR rebased on top of the just released 2.25.0 (https://geoserver.org/announcements/vulnerability/2024/03/19/geoserver-2-25-0-released.html), tests working fine once removed xerces jar.

**Comment by landryb** [comment]
if the block in https://github.com/georchestra/geoserver/blob/2.25.x-georchestra/src/pom.xml#L1372 is dropped, then 2.6.2 seems bundled in the webapp (eg `geoserver/webapp/target/geoserver-generic/WEB-INF/lib/xercesImpl-2.6.2.jar`)

**Comment by f-necas** [comment]
I removed the try catch to match the previous behavior : 
https://github.com/georchestra/geoserver/commit/772f91e4680cda876ace44a7290662aa273c0265

The `try - finally` in the once per reuest filter was a copy of [`org.springframework.security.web.context.SecurityContextPersistenceFilter`.](https://github.com/spring-projects/spring-security/blob/main/web/src/main/java/org/springframework/security/web/context/SecurityContextPersistenceFilter.java)

But the doFilter was overrided in the previous implementation and didn't clear the `SecurityContext` :
https://github.com/turingtestfail/geoserver/blob/6e9e25c0c7cdda9ada9f33f8255130d3afc76801/src/main/src/main/java/org/geoserver/security/filter/GeoServerSecurityContextPersistenceFilter.java

WDYT @landryb @pmauduit ?

**Comment by landryb** [comment]
i don't pretend to fully understand the behaviour of spring internals, but i've tested your commit georchestra/geoserver/@772f91e4680cda876ace44a7290662aa273c0265 on 3 instances (including https://demo.georchestra.org/geoserver/) and that definitely helps seeing the admin menus.

apparently wicket adds `?0=` to the url on the first hit; at that point the admin menus aren't available but dropping `?0=` and reloading the url becomes `?3` and the admin menu is shown, but from what @pmauduit said me that was already the case before.

from my understanding it could/should also be upstreamed, but we're definitely on the right path !

**Comment by landryb** [comment]
rebased the branch on top of master now that #4205 was merged

**Review by f-necas — APPROVED** [review]

**Comment by landryb** [comment]
i see 24.0 was branched, i'll merge this into master once rebased/updated to 2.25.1/retested. I've also just seen that 2.24.3 was released in the meantime...

**Comment by landryb** [comment]
fwiw 2.25.1release includes georchestra/geoserver#36 - i've rebased https://github.com/georchestra/geoserver/commits/2.25.x-georchestra/ on top of latest commit on the 2.25.x branch upstream, eg https://github.com/georchestra/geoserver/commit/6e83128f68fc1b71762fffa68cd22cef64274662

**Comment by landryb** [comment]
runtime-tested ok, connecting as testadmin i reach the admin backend fine, i've been able to use the openlayers preview on a layer (so the xerces impl jar issue above is worked around) and on the modules tab of the server status page i see GT 31.1 & GS 2.25.1 everywhere.

once the GH actions are done i'll merge to master.

**Comment by landryb** [comment]
interesting, OGCAPI tests failing..
```
2024-06-06T10:35:32.5116278Z [INFO] Running org.geoserver.ogcapi.v1.features.CollectionTest
2024-06-06T10:35:33.3372708Z SLF4J(W): No SLF4J providers were found.
2024-06-06T10:35:33.3374023Z SLF4J(W): Defaulting to no-operation (NOP) logger implementation
2024-06-06T10:35:33.3376299Z SLF4J(W): See https://www.slf4j.org/codes.html#noProviders for further details.
2024-06-06T10:35:33.3377846Z SLF4J(W): Class path contains SLF4J bindings targeting slf4j-api versions 1.7.x or earlier.
2024-06-06T10:35:33.3380967Z SLF4J(W): Ignoring binding found at [jar:file:/home/runner/.m2/repository/org/apache/logging/log4j/log4j-slf4j-impl/2.17.2/log4j-slf4j-impl-2.17.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
2024-06-06T10:35:33.3382738Z SLF4J(W): See https://www.slf4j.org/codes.html#ignoredBindings for an explanation.
2024-06-06T10:35:33.7396927Z [ERROR] Tests run: 9, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 1.225 s <<< FAILURE! - in org.geoserver.ogcapi.v1.features.CollectionTest
2024-06-06T10:35:33.7399332Z [ERROR] testDescribeFeatureType(org.geoserver.ogcapi.v1.features.CollectionTest)  Time elapsed: 0.037 s  <<< FAILURE!
2024-06-06T10:35:33.7401057Z java.lang.AssertionError: expected:<200> but was:<500>
2024-06-06T10:35:33.7402711Z 	at org.geoserver.ogcapi.v1.features.CollectionTest.testDescribeFeatureType(CollectionTest.java:287)
2024-06-06T10:35:33.7404054Z 
```
the same `org.geoserver.ogcapi.v1.features.CollectionTest`  test worked in the last run for 2.24.

**Comment by landryb** [comment]
after disabling the failing test in a module we don't apparently enable (ogcapi), the GH actions are green -> merging.

@fvanderbiest : Fwiw there will be GS 2.25.2 & 2.24.4 releases next tuesday, for which i'll prepare pull requests.

**Pull request merged**
